server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -182,5 +198,35 @@ def get_log_path(log_type_param)
           ::Fusor.log_file_path(@deployment.label, @deployment.id)
       end
     end
+
+    def undercloud_handle
+      Overcloud::UndercloudHandle.new('admin', @deployment.openstack_undercloud_password, @deployment.openstack_undercloud_ip_addr, 5000)
+    end
+
+    def get_openstack_param_value(plan, param_name)
+      param = plan.parameters.find { |p| p['name'] == param_name }
+      param['value'] if param
+    end
+
+    def sync_openstack
+      # binding.pry
+      return unless @deployment.deploy_openstack?
+      plan = undercloud_handle.get_plan('overcloud')
+
+      @deployment.openstack_overcloud_ext_net_interface = get_openstack_param_value(plan, 'Controller-1::NeutronPublicInterface')
+      @deployment.openstack_overcloud_libvirt_type = get_openstack_param_value(plan, 'Compute-1::NovaComputeLibvirtType')
+      @deployment.openstack_overcloud_compute_flavor = get_openstack_param_value(plan, 'Compute-1::Flavor')
+      @deployment.openstack_overcloud_compute_count = get_openstack_param_value(plan, 'Compute-1::count')
+      @deployment.openstack_overcloud_controller_flavor = get_openstack_param_value(plan, 'Controller-1::Flavor')
+      @deployment.openstack_overcloud_controller_count = get_openstack_param_value(plan, 'Controller-1::count')
+      @deployment.openstack_overcloud_ceph_storage_flavor = get_openstack_param_value(plan, 'Ceph-Storage-1::Flavor')
+      @deployment.openstack_overcloud_ceph_storage_count = get_openstack_param_value(plan, 'Ceph-Storage-1::Flavor')
+      @deployment.openstack_overcloud_cinder_storage_flavor = get_openstack_param_value(plan, 'Cinder-Storage-1::Flavor')
+      @deployment.openstack_overcloud_cinder_storage_count = get_openstack_param_value(plan, 'Cinder-Storage-1::Flavor')
+      @deployment.openstack_overcloud_swift_storage_flavor = get_openstack_param_value(plan, 'Swift-Storage-1::Flavor')
+      @deployment.openstack_overcloud_swift_storage_count = get_openstack_param_value(plan, 'Swift-Storage-1::Flavor')
```

Comment:
Looks ok to me. Nice way to put some repetitive logic in a method to make it easier to read.